### PR TITLE
feat: add tranche value calculation

### PR DESCRIPTION
**Pull request by marktoda**
This commit adds a function to calculate the value of a tranche in terms
of collateral, even when it is immature

### CODE REVIEW

File: src/entities/bond.ts [hunk 1]
```diff
@@ -90,6 +90,53 @@ export class Bond {
...
+            return CurrencyAmount.fromRawAmount(
+                this.collateral,
+                remainingCollateral
+                    .mul(toBaseUnits(trancheAmount))
+                    .div(zTranche.totalSupply)
+                    .toString(),
+            );
+        }
+
+        // this should never happen given the invariant above
```
Comment (Fiddlekins) [inline src/entities/bond.ts:136]: Currently this can happen when remainingCollateral = 0 for Z tranche and this query is made for the Z tranche balance

Comment (marktoda) [reply src/entities/bond.ts:136]: oh.. true haha

Comment (marktoda) [reply src/entities/bond.ts:136]: should return 0 in that case

Comment (marktoda) [reply src/entities/bond.ts:136]: actually probably dont even need the conditional above... if remaining collateral is 0, we will already return 0

**Review by Fiddlekins — APPROVED** [review]

**Pull request merged**